en/ros2/px4_ros2_interface_lib.md

```diff
@@ -0,0 +1,478 @@
+- when already armed, the mode cannot be selected
+- when armed and the mode is selected, the relevant failsafe is triggered (e.g. RC loss for the manual control requirement).
+  Check the [safety page](../config/safety.md) for how to configure failsafe behavior.
+  A failsafe is also triggered when the mode crashes or becomes unresponsive while it is selected.
+
+This is the corresponding flow diagram for the manual control flag:
+
+![Mode requirements diagram](../../assets/middleware/ros2/px4_ros2_interface_lib/mode_requirements_diagram.png)
+
+<!-- source: https://drive.google.com/file/d/1g_NlQlw7ROLP_mAi9YY2nDwP0zTNsFlB/view -->
+
+It is possible to manually update any mode requirement after the mode is registered.
+For example to add home position as requirement:
+
+```cpp
+modeRequirements().home_position = true;
+```
+
+The full list of flags can be found in [requirement_flags.hpp](https://github.com/Auterion/px4-ros2-interface-lib/blob/main/px4_ros2_cpp/include/px4_ros2/common/requirement_flags.hpp).
+
+#### Deferring Failsafes
+
+A mode or mode executor can temporarily defer non-essential failsafes by calling the method `deferFailsafesSync()`.
+To get notified when a failsafe would be triggered, override the method `void onFailsafeDeferred()`.
+
+Check the [integration test](https://github.com/Auterion/px4-ros2-interface-lib/blob/main/px4_ros2_cpp/test/integration/overrides.cpp) for an example.
+
+### Assigning a Mode to an RC Switch or Joystick Action
+
+External modes can be assigned to [RC switches](../config/flight_mode.md) or joystick actions.
+When assigning a mode to an RC switch, you need to know the index (because the parameter metadata does not contain the dynamic mode name).
+Use `commander status` while the mode is running to get that information.
+
+For example:
+
+```plain
+   INFO  [commander] External Mode 1: nav_state: 23, name: My Manual Mode
+```
+
+means you would select **External Mode 1** in QGC:
+
+![QGC Mode Assignment](../../assets/middleware/ros2/px4_ros2_interface_lib/qgc_mode_assignment.png)
+
+:::note
+PX4 ensures a given mode is always assigned to the same index by storing a hash of the mode name.
+This makes it independent of startup ordering in case of multiple external modes.
+:::
+
+### Replacing an Internal Mode
+
+An external mode can replace an existing internal mode, such as [Return](../flight_modes/return.md) mode (RTL).
+By doing so, whenever RTL gets selected (through the user or a failsafe situation), the external mode is used instead of the internal one.
+The internal one is only used as a fallback when the external one becomes unresponsive or crashes.
+
+The replacement mode can be set in the settings of the `ModeBase` constructor:
+
+```cpp
+Settings{kName, false, ModeBase::kModeIDRtl}
+```
+
+## CI: Integration Tests
+
+When opening a pull request to PX4, CI runs the library integration tests.
+These test that mode registration, failsafes, and mode replacement, work as expected.
```

Comment:
FYI just added a comment about what the tests do now. I know this means possible future maintenance of docs when they do more, but I think it is worth explaining.